vscode-dotnet-sdk-extension/src/extension.ts

```diff
@@ -0,0 +1,150 @@
+        enableTelemetry: enableExtensionTelemetry(extensionConfiguration, configKeys.enableTelemetry),
+        telemetryReporter: extensionContext ? extensionContext.telemetryReporter : undefined,
+    } as IEventStreamContext;
+    const [eventStream, outputChannel, loggingObserver, eventStreamObservers] = registerEventStream(eventStreamContext);
+
+    const displayWorker = extensionContext ? extensionContext.displayWorker : new WindowDisplayWorker();
+    const extensionConfigWorker = new ExtensionConfigurationWorker(extensionConfiguration, undefined);
+    const issueContext = (errorConfiguration: ErrorConfiguration | undefined, commandName: string, version?: string) => {
+        return {
+            logger: loggingObserver,
+            errorConfiguration: errorConfiguration || AcquireErrorConfiguration.DisplayAllErrorPopups,
+            displayWorker,
+            extensionConfigWorker,
+            eventStream,
+            commandName,
+            version,
+        } as IIssueContext;
+    };
+    const timeoutValue = extensionConfiguration.get<number>(configKeys.installTimeoutValue);
+    let storagePath: string;
+    if (os.platform() === 'win32') {
+        storagePath = process.env.APPDATA ? process.env.APPDATA : context.globalStoragePath;
+    } else {
+        storagePath = context.globalStoragePath;
+        if (!fs.existsSync(storagePath)) {
+            fs.mkdirSync(storagePath);
+        }
+    }
+
+    const acquisitionWorker = new DotnetCoreAcquisitionWorker({
+        storagePath,
+        extensionState: context.globalState,
+        eventStream,
+        acquisitionInvoker: new AcquisitionInvoker(context.globalState, eventStream),
+        installationValidator: new InstallationValidator(eventStream),
+        timeoutValue: timeoutValue === undefined ? defaultTimeoutValue : timeoutValue,
+    });
+    const versionResolver = new VersionResolver(context.globalState, eventStream);
+
+    const dotnetAcquireRegistration = vscode.commands.registerCommand(`${commandPrefix}.${commandKeys.acquire}`, async (commandContext: IDotnetAcquireContext) => {
+        const pathResult = callWithErrorHandling(async () => {
+            let version: string | undefined = commandContext ? commandContext.version : undefined;
+            if (!version) {
+                version = await vscode.window.showInputBox({
+                    placeHolder: '5.0',
+                    value: '5.0',
+                    prompt: '.NET version, i.e. 5.0',
+                });
+            }
+            if (!version) {
+                displayWorker.showErrorMessage('No .NET SDK version provided', () => { /* No callback needed */ });
+                return undefined;
+            }
+
+            eventStream.post(new DotnetAcquisitionRequested(version!));
+            const resolvedVersion = await versionResolver.getFullSDKVersion(version!);
+            const dotnetPath = await acquisitionWorker.acquireSDK(resolvedVersion);
```

Comment:
The resolve existing path API doesn't seem to be used here. Just checking, acquireSDK won't try and re-download an already acquired SDK still correct?

Reply:
Yes, the resolve existing path logic had to do with the `[existingDotnetPath`](https://github.com/dotnet/vscode-dotnet-runtime/pull/163/files#diff-56eb173bf3821e528c6cb0129059298ebb627e3a9864955269f63684011d4fc1R59) configuration value, not the SDK we acquire.